pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -0,0 +1,150 @@
+    checksumtype = models.TextField()
+
+    repodata = JSONField(dict)
+    is_modular = models.BooleanField(default=False)
+    size = models.PositiveIntegerField()
+    filename = models.TextField()
+
+    pulp2_type = 'rpm'
+
+    class Meta:
+        unique_together = (
+            'name', 'epoch', 'version', 'release', 'arch', 'checksumtype', 'checksum',
+            'pulp2content')
+        default_related_name = 'rpm_detail_model'
+
+    @property
+    def expected_digests(self):
+        """Return expected digests."""
+        return {self.checksumtype: self.checksum}
+
+    @property
+    def expected_size(self):
+        """Return expected size."""
+        return self.size
+
+    @property
+    def relative_path_for_content_artifact(self):
+        """Return relative path."""
+        return self.filename
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 RPM content with all the fields needed to create a Pulp 3 Package.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+        pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_rpm_content_batch = RPM.objects.filter(id__in=pulp2_ids)
+        pulp2rpm_to_save = [
+            cls(name=rpm.name,
+                epoch=rpm.epoch,
+                version=rpm.version,
+                release=rpm.release,
+                arch=rpm.arch,
+                checksum=rpm.checksum,
+                checksumtype=rpm.checksumtype,
+                repodata=rpm.repodata,
+                is_modular=rpm.is_modular,
+                size=rpm.size,
+                filename=rpm.filename,
+                pulp2content=pulp2_id_obj_map[rpm.id])
+            for rpm in pulp2_rpm_content_batch]
+        cls.objects.bulk_create(pulp2rpm_to_save, ignore_conflicts=True)
+
+
+class Pulp2Erratum(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store Pulp2 Errata content details.
+    """
+
+    # Required fields
+    errata_id = models.TextField(unique=True)
+    updated = models.TextField()
+
+    issued = models.TextField()
+    status = models.TextField()
+    description = models.TextField()
+    pushcount = models.TextField()
+    references = JSONField()
+    reboot_suggested = models.BooleanField()
+    relogin_suggested = models.BooleanField()
+    restart_suggested = models.BooleanField()
+    errata_from = models.TextField()
+    severity = models.TextField()
+    rights = models.TextField()
+    version = models.TextField()
+    release = models.TextField()
+    errata_type = models.TextField()
+    pkglist = JSONField()
+    title = models.TextField()
+    solution = models.TextField()
+    summary = models.TextField()
+
+    pulp2_type = 'erratum'
+
+    class Meta:
+        default_related_name = 'erratum_detail_model'
```

Comment:
@goosemania we need to add here properties `expected_size, expected_digests, relative_path_for_content_artifact` because later code would fail whenever it will try to access those

Reply:
+1  I hoped errata migration code will take care of that, since I can't test errata now.